networkit/cpp/graph/GraphTools.cpp

```diff
@@ -201,14 +201,65 @@ double density(const Graph &G) noexcept {
 }
 
 double volume(const Graph &G) {
-    if (G.isDirected())
-        throw std::runtime_error("Volume computation is only supported for undirected graphs.");
+    auto volume = 0.0;
 
-    if (!G.isWeighted()) {
-        return 2.0 * G.numberOfEdges();
+    if (G.isWeighted()) {
+        volume = G.totalEdgeWeight();
+    } else {
+        volume = G.numberOfEdges();
+    }
+    if (!G.isDirected()) {
+        volume *= 2.0;
+    }
+    return volume;
+}
+
+double subgraphInVolume(const Graph &G, const std::unordered_set<node> &nodes) {
+    double inVolume = 0.0;
+    bool countSelfLoopsTwice = !G.isDirected();
+    count nodesNotInGraph = 0;
+
+    // Check for nodes in subset, which are not part of the graph
+    for (auto it = std::begin(nodes); it != std::end(nodes); ++it) {
```

Comment:
What about `for (node u : nodes)`?

Reply:
Of course my comment does not hold anymore if we are going to use ranges for the input nodes.